src/ExpressiveInstaller/OptionalPackages.php

```diff
@@ -172,16 +198,52 @@ public static function install(Event $event)
         // Update composer definition
         $json->write(self::$composerDefinition);
 
-        // Minimal install? Remove default middleware
-        if ($minimal) {
-            self::removeDefaultMiddleware($io, self::$projectRoot);
-        }
-
         self::clearComposerLockFile($io, self::$projectRoot);
         self::cleanUp($io, self::$projectRoot);
     }
 
     /**
+     * @param IOInterface $io
+     * @param string $installType
+     * @param array $appFiles Values from the 'application' key
+     * @return void
+     * @throws RuntimeException if $installType is unknown
+     */
+    public static function setupDefaultApp(IOInterface $io, $installType, array $appFiles)
+    {
+        switch ($installType) {
+            case self::INSTALL_MINIMAL:
+                self::removeDefaultModule($io, self::$projectRoot);
+                // no files to copy
+                return;
+
+            case self::INSTALL_FLAT:
+                // Re-arrange files into a flat structure.
+                self::recursiveRmdir(self::$projectRoot . '/src/App/templates');
+                rename(self::$projectRoot . '/src/App/src/Action', self::$projectRoot . '/src/App/Action');
+                self::recursiveRmdir(self::$projectRoot . '/src/App/src');
+
+                // Re-define autoloading rules
+                self::$composerDefinition['autoload']['psr-4']['App\\'] = 'src/App/';
```

Comment:
The current skeleton defines the tree:

```text
src/
    App/
        Action/
            HomePageAction.php
            HomePageActionFactory.php
            PingAction.php
```

As such, "flat" is meant to mimic that structure; it's essentially marking `src/` as a PSR-0 root directory, if that makes sense.